journal/src/main/java/io/camunda/zeebe/journal/file/Segment.java

```diff
@@ -277,4 +277,18 @@ private void markForDeletion() {
     }
     markedForDeletion = true;
   }
+
+  void updateDescriptor() {
+    descriptor.setLastIndex(writer.getLastIndex());
+    descriptor.setLastPosition(writer.getLastEntryPosition());
+    descriptor.updateIfCurrentVersion(buffer);
+  }
+
+  void resetLastEntryInDescriptor() {
+    descriptor.setLastIndex(0);
+    descriptor.setLastPosition(0);
+    descriptor.updateIfCurrentVersion(buffer);
+    // flush immediately to prevent inconsistencies between descriptor and actual last written entry
+    buffer.force(0, descriptor.length());
```

Comment:
This doesn't reduce the guarantees, right? It might flush more often than we expected. So in that sense it is not reducing the guarantee. 
Besides, this is only done during truncation, which will not happen often.

Reply:
True. It could be confusing. But I think we also force flush when we take snapshot, overriding the configured flush. Sometimes it is necessary to prevent inconsistencies.